src/Analyzer.Core/TemplateAnalyzer.cs

```diff
@@ -114,42 +166,93 @@ public IEnumerable<IEvaluation> AnalyzeTemplate(string template, string template
+                SourceMap = parentContext.SourceMap,
+                PathPrefix = pathPrefix,
+                ParentContext = parentContext
             };
 
             try
             {
                 IEnumerable<IEvaluation> evaluations = this.jsonRuleEngine.AnalyzeTemplate(templateContext);
                 evaluations = evaluations.Concat(this.powerShellRuleEngine.AnalyzeTemplate(templateContext));
 
-                // For each rule we don't want to report the same line more than once
-                // This is a temporal fix
-                var evalsToValidate = new List<IEvaluation>();
-                var evalsToNotValidate = new List<IEvaluation>();
-                foreach (var eval in evaluations)
+                // Recursively handle nested templates 
+                var jsonTemplate = JObject.Parse(populatedTemplate);
+                var processedTemplateResources = templatejObject.InsensitiveToken("resources");
+
+
+                for (int i = 0; i < processedTemplateResources.Count(); i++)
                 {
-                    if (!eval.Passed && eval.Result != null)
-                    {
-                        evalsToValidate.Add(eval);
-                    }
-                    else
+                    var currentProcessedResource = processedTemplateResources[i];
+
+                    if (currentProcessedResource.InsensitiveToken("type").ToString().Equals("Microsoft.Resources/deployments", StringComparison.OrdinalIgnoreCase))
                     {
-                        evalsToNotValidate.Add(eval);
+                        var nestedTemplate = currentProcessedResource.InsensitiveToken("properties.template");
+                        if (nestedTemplate == null)
+                        {
+                            continue; // This is a Linked template 
+                        }
+                        var populatedNestedTemplate = nestedTemplate.DeepClone();
+
+                        // Check whether scope is set to inner or outer
+                        var scope = currentProcessedResource.InsensitiveToken("properties.expressionEvaluationOptions")?.InsensitiveToken("scope")?.ToString();
+                        //string nextPathPrefix = $".properties.template.";
+                        string nextPathPrefix = nestedTemplate.Path;
+                       
+                        IEnumerable<IEvaluation> result;
+
+                        if (scope == null || scope == "outer")
+                        {
+                            // Variables, parameters and functions inherited from parent template
+                            string functionsKey = populatedNestedTemplate.InsensitiveToken("functions")?.Parent.Path;
+                            string variablesKey = populatedNestedTemplate.InsensitiveToken("variables")?.Parent.Path;
+                            string parametersKey = populatedNestedTemplate.InsensitiveToken("parameters")?.Parent.Path;
```

Comment:
I'm a little confused here.  I know for outer scope the template will share everything from the parent, but why only copy the parent data if the nested property names already exist in the nested template?  If they do already exist, what's in them?  Wouldn't that end up disappearing when you copy the parent data into it?

Reply:
Yeah, they do. In outer scope they shouldn't be evaluated so they never really get used. Which is why I am ovewriting them

Reply:
Outer scoped templates should always have a reference to these in the parent though, right?  It seems these are only copied into the nested template if the nested template tries to define its own, which isn't allowed anyway.  So if the template does the right thing and doesn't define its own, it won't get a reference to the parent items?

Reply:
not really, what I do is assign the parent parameters, variables ... to the child template regardless. I don't make any checks for outer scope so they always get the parent parameters, variables and functions

Reply:
We might be thinking different things.

Take functions for example.  Functions are assigned on line 216, right?  That only happens if `functionsKey` (the functions in the nested template, from line 210) is not null, meaning the nested template defines functions.  All of this is after checking that the scope is "outer" on line 207.  So for an outer scope that *doesn't* define functions, they aren't copied.  Am I missing something somewhere?

Reply:
Oh! (facepalm)
You're right! The implementation does what you are saying after I made the change since parameters, variables and functions don't get passed if the key is null. 
Let me fix that. Sorry for the confusion, I hadn't seen that myself.

Reply:
This means there is quite a large gap in tests.  There should be tests that verify functions, variables, and parameters are made available to outer scoped templates.  Tests like that should be able to catch things like this.